### PR TITLE
chore: add prow OWNERS file

### PR DESCRIPTION
**Pull request by wuhuizuo**
Ref: #1230

Why: we need control the changes of protobuf files strictly, the changes should approved by suitable teams.

- [x] Also need a repository level OWNERS file that filled with default reviewers and approvers.
- [x] Setup OWNERS for `proto/` folder.
   > Who will be the approvers for the following files:
  - [x] proto/autoid.proto @bb7133 @tiancaiamao 
  - [x] proto/brpb.proto @BornChanger @overvenus
  - [x] proto/cdcpb.proto @flowbehappy @overvenus
  - [x] proto/configpb.proto @overvenus
  - [x] proto/coprocessor.proto @cfzjywxk
  - [x] proto/deadlock.proto @MyonKeminta @cfzjywxk 
  - [x] proto/debugpb.proto @cfzjywxk 
  - [ ] proto/diagnosticspb.proto
  - [ ] proto/disaggregated.proto
  - [x] proto/disk_usage.proto @tonyxuqqi @overvenus 
  - [x] proto/encryptionpb.proto @tonyxuqqi @overvenus 
  - [x] proto/enginepb.proto @overvenus @tonyxuqqi
  - [x] proto/errorpb.proto @cfzjywxk 
  - [ ] proto/gcpb.proto
  - [x] proto/import_kvpb.proto @BornChanger @D3Hunter @lance6716
  - [x] proto/import_sstpb.proto @BornChanger @D3Hunter @lance6716
  - [x] proto/keyspacepb.proto @niubell @rleungx
  - [x] proto/kvrpcpb.proto @cfzjywxk 
  - [x] proto/logbackuppb.proto @BornChanger 
  - [x] proto/metapb.proto @tonyxuqqi @overvenus 
  - [x] proto/meta_storagepb.proto @niubell @rleungx
  - [ ] proto/mpp.proto
  - [x] proto/pdpb.proto @niubell @rleungx
  - [x] proto/raft_cmdpb.proto @tonyxuqqi @overvenus 
  - [x] proto/raft_serverpb.proto @tonyxuqqi @overvenus 
  - [x] proto/recoverdatapb.proto @BornChanger 
  - [ ] proto/replication_modepb.proto
  - [x] proto/resource_manager.proto @tonyxuqqi @overvenus 
  - [ ] proto/resource_usage_agent.proto
  - [x] proto/schedulingpb.proto @niubell @rleungx
  - [x] proto/tikvpb.proto @tonyxuqqi  @cfzjywxk 
  - [x] proto/tracepb.proto @you06 
  - [x] proto/tsopb.proto @niubell @rleungx

### CODE REVIEW

**Comment by wuhuizuo** [comment]
Who will be the approvers for the following files:

- [ ] proto/autoid.proto
- [ ] proto/brpb.proto
- [ ] proto/cdcpb.proto
- [ ] proto/configpb.proto
- [ ] proto/coprocessor.proto
- [ ] proto/deadlock.proto
- [ ] proto/debugpb.proto
- [ ] proto/diagnosticspb.proto
- [ ] proto/disaggregated.proto
- [ ] proto/disk_usage.proto
- [ ] proto/encryptionpb.proto
- [ ] proto/enginepb.proto
- [ ] proto/errorpb.proto
- [ ] proto/gcpb.proto
- [ ] proto/import_kvpb.proto
- [ ] proto/import_sstpb.proto
- [ ] proto/keyspacepb.proto
- [ ] proto/kvrpcpb.proto
- [ ] proto/logbackuppb.proto
- [ ] proto/metapb.proto
- [ ] proto/meta_storagepb.proto
- [ ] proto/mpp.proto
- [ ] proto/pdpb.proto
- [ ] proto/raft_cmdpb.proto
- [ ] proto/raft_serverpb.proto
- [ ] proto/recoverdatapb.proto
- [ ] proto/replication_modepb.proto
- [ ] proto/resource_manager.proto
- [ ] proto/resource_usage_agent.proto
- [ ] proto/schedulingpb.proto
- [ ] proto/tikvpb.proto
- [ ] proto/tracepb.proto
- [ ] proto/tsopb.proto

**Comment by wuhuizuo** [comment]
@bb7133 will @tiancaiamao be a code owner for `proto/autoid.proto`? any others?

**Comment by BornChanger** [comment]
`proto/recoverdatapb.proto` is also in BR domain. @wuhuizuo

**Comment by tonyxuqqi** [comment]
raftstore related proto files:
proto/disk_usage.proto
proto/encryptionpb.proto
proto/metapb.proto
proto/raft_cmdpb.proto
proto/raft_serverpb.proto
proto/resource_manager.proto

**Comment by tonyxuqqi** [comment]
@wuhuizuo Can multiple people be the owner of some files? If yes, please add @overvenus as owner as well for raftstore related proto files

**Comment by wuhuizuo** [comment]
> @wuhuizuo Can multiple people be the owner of some files? If yes, please add @overvenus as owner as well for raftstore related proto files

Yes, I will update it.

**Comment by cfzjywxk** [comment]
@wuhuizuo 
I made some changes to the description updateing owner information, PTAL

**Comment by wuhuizuo** [comment]
@tonyxuqqi could you help me to solve the failure of the `rust-test` CI task? It looks it deals all files include the `OWNERS` file.

File: OWNERS
Comment (wuhuizuo) [inline OWNERS:3]:
```suggestion
  - sig-approvers

```

File: scripts/check.sh
Comment (wuhuizuo) [inline scripts/check.sh:27]:
```suggestion
```

**Review by tonyxuqqi — APPROVED** [review]

**Comment by wuhuizuo** [comment]
/cc @bb7133 @tiancaiamao @BornChanger @flowbehappy @cfzjywxk @MyonKeminta @cfzjywxk @tonyxuqqi @overvenus @niubell @you06 @lance6716 @YuJuncen

PTAL

**Review by lance6716 — APPROVED** [review]

**Comment by wuhuizuo** [comment]
/hold
wait for more review approvals.

**Review by cfzjywxk — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

File: OWNERS_ALIASES
Comment (wuhuizuo) [inline OWNERS_ALIASES:23]:
```suggestion
  sig-approvers-pb-pd: [niubell]
  sig-approvers-pb-trace: [you06]
```

File: OWNERS_ALIASES
Comment (wuhuizuo) [inline OWNERS_ALIASES:19]:
```suggestion
    sig-approvers-pb-pd: [niubell]
    sig-approvers-pb-raftstore: [tonyxuqqi, overvenus]
```

File: proto/OWNERS
Comment (wuhuizuo) [inline proto/OWNERS:64]:
```suggestion
  "^pdpb\\.proto$":
    approvers: [sig-approvers-pb-pd]
 
  ### If the special file need to be controlled by a SIG, please uncomment it and fill the approvers.
  # "^configpb\\.proto$":
  #   approvers: []
  # "^diagnosticspb\\.proto$":
  #   approvers: []
  # "^disaggregated\\.proto$":
  #   approvers: []
  # "^enginepb\\.proto$":
  #   approvers: []
  # "^gcpb\\.proto$":
  #   approvers: []
  # "^keyspacepb\\.proto$":
  #   approvers: []
  # "^meta_storagepb\\.proto$":
  #   approvers: []
  # "^mpp\\.proto$":
  #   approvers: []
```

**Review by flowbehappy — APPROVED** [review]

File: OWNERS_ALIASES
Comment (wuhuizuo) [inline OWNERS_ALIASES:19]:
```suggestion
  sig-approvers-pb-pd: [niubell, rleungx]
```

**Review by you06 — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

File: proto/OWNERS
Comment (wuhuizuo) [inline proto/OWNERS:72]:
```suggestion
  "^keyspacepb\\.proto$":
    approvers: [sig-approvers-pb-pd]
  "^schedulingpb\\.proto$":
    approvers: [sig-approvers-pb-pd]
  "^tsopb\\.proto$":
    approvers: [sig-approvers-pb-pd]
  
  ### If the special file need to be controlled by a SIG, please uncomment it and fill the approvers.
  # "^configpb\\.proto$":
  #   approvers: []
  # "^diagnosticspb\\.proto$":
  #   approvers: []
  # "^disaggregated\\.proto$":
  #   approvers: []
  # "^enginepb\\.proto$":
  #   approvers: []
  # "^gcpb\\.proto$":
  #   approvers: []
  # "^meta_storagepb\\.proto$":
  #   approvers: []
  # "^mpp\\.proto$":
  #   approvers: []
  # "^replication_modepb\\.proto$":
  #   approvers: []
  # "^resource_usage_agent\\.proto$":
  #   approvers: []
```

**Review by niubell — APPROVED** [review]

File: scripts/check.sh
Comment (overvenus) [inline scripts/check.sh:23]: By removing `GO111MODULE=off`, please increase the go requirement to `1.16` in go.mod.
https://go.dev/ref/mod#:~:text=Since%20Go%201.16%2C%20go%20install%20is%20the%20recommended%20command

File: build.rs
Comment (overvenus) [inline build.rs:35]: Why remove `search_dir_for_protos`?

Comment (wuhuizuo) [reply build.rs:35]: `search_dir_for_protos` will include all files, the implement is not considered to exclude the no-proto files, it made the `make rust` failed.

and we add the OWNERS to this dir for best practice.

Comment (wuhuizuo) [reply build.rs:35]: ignore the no proto files in the commit:
https://github.com/pingcap/kvproto/pull/1231/commits/d9846b048e0156ec355a5e61252bd96ce6032920

**Comment by overvenus** [comment]
Can I be granted co-ownership of the `brpb.proto`, `cdcpb.proto`, `configpb.proto` and `enginepb.proto` files? I have written a significant portion of their content and the lessons learned from their development.

**Comment by wuhuizuo** [comment]
> Can I be granted co-ownership of the `brpb.proto`, `cdcpb.proto`, `configpb.proto` and `enginepb.proto` files? I have written a significant portion of their content and understand the lessons learned from their development.

PTAL 
- @BornChanger `brpb.proto`
- @flowbehappy `cdcpb.proto`

**Comment by BornChanger** [comment]
> Can I be granted co-ownership of the `brpb.proto`, `cdcpb.proto`, `configpb.proto` and `enginepb.proto` files? I have written a significant portion of their content and understand the lessons learned from their development.
Sure.

**Review by flowbehappy — APPROVED** [review]

**Comment by rleungx** [comment]
proto/meta_storagepb.proto belongs to PD.

**Comment by wuhuizuo** [comment]
updated in https://github.com/pingcap/kvproto/pull/1231/commits/dd951a08870d53041c6388ff632ff98441a41bc7
@rleungx @overvenus @flowbehappy @BornChanger PTAL

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by MyonKeminta — APPROVED** [review]

**Comment by wuhuizuo** [comment]
/cc @Benjamin2037 
PTAL 16101f477272b29eb1bb9a47f4a399e3183f3c2f

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**
